pkg/orchestrator/selection/ranking/available_capacity_test.go

```diff
@@ -0,0 +1,292 @@
+//go:build unit || !integration
+
+package ranking
+
+import (
+	"context"
+	"sort"
+	"testing"
+
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+	"github.com/bacalhau-project/bacalhau/pkg/test/mock"
+	"github.com/stretchr/testify/suite"
+)
+
+type AvailableCapacityNodeRankerSuite struct {
+	suite.Suite
+	ranker *AvailableCapacityNodeRanker
+}
+
+func (suite *AvailableCapacityNodeRankerSuite) SetupTest() {
+	suite.ranker = NewAvailableCapacityNodeRanker()
+}
+
+type nodeScenario struct {
+	nodeID            string
+	availableCapacity models.Resources
+	queueUsedCapacity models.Resources
+}
+
+type rankNodesTestCase struct {
+	name         string
+	nodes        []nodeScenario
+	jobResources models.ResourcesConfig
+	expected     []string
+	equalCheck   bool
+}
+
+func (suite *AvailableCapacityNodeRankerSuite) TestRankNodes() {
```

Comment:
I'd find this test _slightly_ more readable if the node resources were scaled from a common value, e.g.
But don't feel strongly enough to push for change
```go
bc := models.Resources{CPU: 4, Memory: 16000, Disk: 200000, GPU: 1}
bq :=  models.Resources{CPU: 1, Memory: 4000, Disk: 100000, GPU: 0},
nodes: []nodeScenario{
	{
		nodeID:            "node1",
		availableCapacity: bc,
		queueUsedCapacity: bq,
	},
	{
		nodeID:            "node2",
		availableCapacity: models.Resources{CPU: bc.CPU*2, Memory: bc.Memory*2, Disk: bc.Disk/2, GPU: bc.GPU*2},
		queueUsedCapacity: models.Resources{CPU: bq.CPU/2, Memory: bq.Memory/2, Disk: bc.Disk/2, GPU: bc.GPU},
	},
	{
		nodeID:            "node3",
		availableCapacity: models.Resources{CPU: bc.CPU/2, Memory: bc.Memory, Disk: bc.Disk-5000, GPU: bc.GPU*4},
		queueUsedCapacity: models.Resources{CPU: bq.CPU/2, Memory: bq.Memory/4, Disk: bq.Disk/10, GPU: bq.GPU*2},
	},
},
```